src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/UnconfiguredProjectContextProvider.cs

```diff
@@ -214,15 +216,18 @@ private void AddConfiguredProjectState(ConfiguredProject configuredProject, IWor
             if (string.IsNullOrEmpty(targetPath))
                 return null;
 
-
-            // TODO: https://github.com/dotnet/roslyn-project-system/issues/353
-            await _commonServices.ThreadingService.SwitchToUIThread();
+            bool continueOnCaturedContext = false;
```

Comment:
nit: misspelling. I think you meant `continueOnCapturedContext`

Reply:
Thanks, fixed (rebased into original commit)